apple/internal/resource_rules/apple_core_ml_library.bzl

```diff
@@ -153,8 +153,76 @@ as `#import my/package/MyModel.h`.
 
 This rule will also compile the `mlmodel` into an `mlmodelc` and propagate it
 upstream so that it is packaged as a resource inside the top level bundle.
+""",
+)
+
+def _swift_apple_core_ml_library_impl(ctx):
```

Comment:
Looking at the differences between this and the existing `_apple_core_ml_library_impl` it seems like there are very few changes. instead of creating a separate rule for this what do you think about having a parameter on that internal rule for whether it generates swift or not, I guess just something like `generates_swift = True`, defaulting to false for keeping the current behavior. And then doing the same in the macro, to either create a objc_library or swift_library as the final target.

I think overall that would be a nicer user interface, and you'd dodge your naming concern. Thoughts?

Reply:
I had an implementation like this in one of my commits on this branch. I wasn't able to make it work in practice, though, due to a larger, critical detail between `outputs = { "source": "%{name}.swift" }` (in the Swift version) and `outputs = { "source": "%{name}.m" }` (in the Obj-C version).

is there a way to declare a rule that supports something like this? Otherwise I agree that having a "language" parameter, similar to the `apple_intent_library` rule, would be ideal.

Reply:
If I understand the use case you don't have to predeclare outputs like this (it's actually discouraged now). You can just call ctx.actions.declare_file with the name instead.

Reply:
Interesting! I'll take another look tomorrow then 😃